hnn_core/parallel_backends.py

```diff
@@ -671,7 +742,7 @@ def __exit__(self, type, value, traceback):
         _BACKEND = self._old_backend
 
         # always kill nrniv processes for good measure
-        if self.n_procs > 1:
+        if (not self.mpi_comm_spawn) and self.n_procs > 1:
             kill_proc_name('nrniv')
```

Comment:
Note that `kill_proc_name` only works when a subprocess is started with `Popen` and appended to `proc_queue`.